frontend/packages/console-app/src/components/nodes/NodeLogs.tsx

```diff
@@ -144,113 +257,110 @@ const NodeLogs: React.FC<NodeLogsProps> = ({ obj: node }) => {
-    setLogFilenames([]);
-    setLogFilename('');
-    setUnit('');
-    setQueryArgument(pathQueryArgument, newAPI);
-    removeQueryArgument(unitQueryArgument);
-    removeQueryArgument(logQueryArgument);
-    setLoadingFilenames(true);
-    setLoadingLog(true);
-    trimmedContent = '';
-  };
-  const onTogglePath = () => setPathOpen(!isPathOpen);
-  const onChangeUnit = (value: string) => {
-    setUnit(value);
-    value === ''
-      ? removeQueryArgument(unitQueryArgument)
-      : setQueryArgument(unitQueryArgument, value);
-  };
-  const onChangeFilename = (event: React.ChangeEvent<HTMLInputElement>, newFilename: string) => {
-    event.preventDefault();
-    setFilenameOpen(false);
-    setLogFilename(newFilename);
-    setLoadingLog(true);
-    setQueryArgument(logQueryArgument, newFilename);
-    setLogURL(getLogURL(`/${newFilename}`));
-    fetchLog(getLogURL(`/${newFilename}`));
-    trimmedContent = '';
-  };
-  const onToggleFilename = () => setFilenameOpen(!isFilenameOpen);
-  const options = (items) =>
-    items.map((value) => {
-      return (
-        <SelectOption
-          key={value}
-          value={value}
-          className={classnames({ 'co-node-logs__log-select-option': value.length > 50 })}
-        />
-      );
-    });
-  const logLabel = t('public~Select a log file');
+  const onChangePath = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>, newAPI: string) => {
+      event.preventDefault();
+      setPathOpen(false);
+      setPath(newAPI);
+      setLogFilenames([]);
+      setLogFilename('');
+      setUnit('');
+      setQueryArgument(pathQueryArgument, newAPI);
+      removeQueryArgument(unitQueryArgument);
+      removeQueryArgument(logQueryArgument);
+      setLoadingFilenames(true);
+      setLoadingLog(true);
+      trimmedContent.current = '';
+    },
+    [pathQueryArgument, unitQueryArgument, logQueryArgument],
+  );
+  const onTogglePath = React.useCallback(() => setPathOpen(!isPathOpen), [isPathOpen]);
+  const onChangeUnit = React.useCallback(
+    (value: string) => {
+      setUnit(value);
+      value === ''
+        ? removeQueryArgument(unitQueryArgument)
+        : setQueryArgument(unitQueryArgument, value);
+    },
+    [unitQueryArgument],
+  );
+  const onChangeFilename = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>, newFilename: string) => {
+      event.preventDefault();
+      setFilenameOpen(false);
+      setLogFilename(newFilename);
+      setLoadingLog(true);
+      setQueryArgument(logQueryArgument, newFilename);
+      setLogURL(getLogURL(`/${newFilename}`));
+      fetchLog(getLogURL(`/${newFilename}`));
+      trimmedContent.current = '';
+    },
+    [fetchLog, getLogURL],
+  );
+  const onToggleFilename = React.useCallback(() => setFilenameOpen(!isFilenameOpen), [
+    isFilenameOpen,
+  ]);
   const errorExists = error.length > 0;
   const logFilenamesExist = logFilenames.length > 0;
 
+  const logControls = React.useMemo(
```

Comment:
I dont think there is any benefit in using `React.useMemo` in this case

Reply:
Won't the component unmount/remount every render otherwise, or am I wrong? Maybe not terrible since it's stateless.

Reply:
I dont think it will. It should be decided by React's reconciliation cycle. I think this is the same as passing children. We never memoize children JSX so I dont see a reason to do it here.

If removing useMemo causes some unnecessary rerenders and its causing some trouble I think a better solution is to wrap `LogControls` component into React.memo

Reply:
Looking again, I believe I'm wrong. You can remove `useMemo`.